contracts/access/IAccessControlAdminRules.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.9.0 (access/IAccessControlAdminRules.sol)
+
+pragma solidity ^0.8.0;
+
+import "./IAccessControl.sol";
+
+/**
+ * @dev External interface of AccessControladminRules declared to support ERC165 detection.
+ *
+ * _Available since v4.9._
+ */
+interface IAccessControlAdminRules is IAccessControl {
+    /**
+     * @dev Emitted when an `DEFAULT_ADMIN_ROLE` transfer is started, setting `newAdmin`
+     * as the next owner to be claimed after `delayedUntil` is met.
+     */
+    event AdminRoleChangeStarted(address indexed newAdmin, uint48 delayedUntil);
+
+    /**
+     * @dev Returns the address of the current `DEFAULT_ADMIN_ROLE` owner.
+     */
+    function owner() external view returns (address);
+
+    /**
+     * @dev Returns the timestamp in which the pending admin can claim the
+     * `DEFAULT_ADMIN_ROLE` role.
+     */
+    function delayedUntil() external view returns (uint48);
+
+    /**
+     * @dev Returns the address of the pending `DEFAULT_ADMIN_ROLE` owner.
+     */
+    function pendingAdmin() external view returns (address);
+
+    /**
+     * @dev Starts a `DEFAULT_ADMIN_ROLE` transfer by setting a pending admin and
+     * a timer to be met.
+     *
+     * Requirements:
+     *
+     * - There shouldn't be another admin transfer in progress. See {cancelAdminTransfer}.
+     * - Can only be called by the current `DEFAULT_ADMIN_ROLE` owner.
+     *
+     * Emits an {AdminRoleChangeStarted}.
+     */
+    function beginAdminTransfer(address newAdmin) external;
+
+    /**
+     * @dev Completes a `DEFAULT_ADMIN_ROLE` transfer.
+     *
+     * Requirements:
+     *
+     * - Caller should be the pending admin.
+     * - `DEFAULT_ADMIN_ROLE` should be granted to the caller.
+     * - `DEFAULT_ADMIN_ROLE` should be revoked from the previous owner.
+     * - Should allow to call {beginAdminTransfer} again.
+     */
+    function acceptAdminTransfer() external;
+
+    /**
+     * @dev Cancels a `DEFAULT_ADMIN_ROLE` transfer.
+     *
+     * Requirements:
+     *
+     * - Should allow to call {beginAdminTransfer} again.
+     * - Can be called even after the timer is met.
+     * - Can only be called by the current `DEFAULT_ADMIN_ROLE` owner.
+     */
+    function cancelAdminTransfer() external;
```

Comment:
Again, I don't see the point of having a `cancelAdminTransfer` in addition of `beginAdminTransfer`

Right now, we can:
- begin a transfer
- cancel
- re-initiate a transfer

So the inability to call begin when the transfer when another one is in progress does not really exist, we are just forcing calling cancel first, which IMO is just an inefficiency.

I would personally remove the `cancelAdminTransfer` function, and allow `beginAdminTransfer` to override a pending transfer.

Reply:
I'd love to do this but I see no clear path for handling the `address(0)` case which is needed for renouncing, so we can't use it as a way to cancel.

Only way I see is using the `delayUntil()` as a flag

```solidity
    function beginAdminTransfer(address newAdmin) public override onlyRole(DEFAULT_ADMIN_ROLE) {
        if(delayedUntil() == 0) {
            _delayedUntil = SafeCast.toUint48(block.timestamp) + _delay;
            _pendingAdmin = newAdmin;
            emit AdminRoleChangeStarted(_pendingAdmin, _delayedUntil);
        } else {
            _resetAdminTransfer();
        }
    }
```

But I'm not convinced

Reply:
you did not get what I'm saying. 

I would do
```solidity
function beginAdminTransfer(address newAdmin) public override onlyRole(DEFAULT_ADMIN_ROLE) {
        _delayedUntil = SafeCast.toUint48(block.timestamp) + _delay;
        _pendingAdmin = newAdmin;
        emit AdminRoleChangeStarted(_pendingAdmin, _delayedUntil);
    }
```
and remove the cancel function.

If you want to cancel, just call `beginAdminTransfer(0x1)` and it won't be possible to accept.

Reply:
I don't get what I'm supposed to get, then. I don't think address(1) (or any other non-zero) for cancelling is the best, I think is misleading and may require analyzing its effect with precompiles (ecrecover it's safe but I don't want to add that complexity) while making recommendations on how to cancel.

Also, `begin*` for canceling is misleading.

I'd keep the cancel.

Reply:
> If you want to cancel, just call `beginAdminTransfer(0x1)` and it won't be possible to accept.

This isn't a good option. I don't see why we wouldn't want the cancel function.

Reply:
I got the length from both deployment bytecodes and this is the result:

- With cancel function: 834564
- Without the cancel: 825345 **(-1.104%)**

Using `beginAdminTransfer` would be more costly than just canceling. I don't think the deployment cost savings are worth it.

Reply:
Copying the reply from another thread, not sure why it was there:

>I'm ok to keep the `cancelAdminTransfer` for workflow were the cancel is not preceding a `beginAdminTransfer`.
>
>Said otherwize: yes, lets keep the cancel function, but lets also remove the `require(delayedUntil() == 0, "AccessControl: pending admin already set");` check in `beginAdminTransfer`

Reply:
Keeping both sounds good to me. Although I'd mention this implies that the `defaultAdmin` can front-run an `acceptDefaultAdminTransfer` call from the pending `defaultAdmin`, which I'd say is expected.